staffdocs: Add getinvolved page (#455)
ocf/ocfweb: ocfweb/docs/docs/staff/getinvolved.md

﻿[[!meta title="Getting Involved"]]

## How can I get involved in the OCF?

Glad you asked! Students from all backgrounds are welcome to join staff. We

> don’t interview or have a selective process for joining-- you simply have to

Reviewer: eM dAsH pLz

express interest and show up to our weekly meetings. If you want to learn more
and be an active participant, there are multiple ways to get started:

### Stay up-to-date

We use a mailing list ([ocf.io/announce][announce]) to announce weekly meetings
and recap past meetings. If you’re joining in the middle of the semester or
missed a meeting, you can also read the mailing list archives to see what we
went over in the past. This is a low volume mailing list (2 emails per week)
and you can unsubscribe at any time.

[[Read more about other mailing lists available to join|doc
staff/mailing-lists]]

### In-person meetings

We hold two meetings every Monday:

#### Board of Directors (BoD)

The OCF Board of Directors has open meetings from 7-8 pm every Monday. BoD is
where we discuss and vote on administrative decisions. If you are interested in
gaining insight into the internal operations the OCF from an administrative

> perspective, please come to BoD meetings! Meetings always are open to all.

Reviewer 2: I think that there should be more description on what happens at BoD meetings, to attract people potentially more into the administrative side of the OCF. I think there should be a bit longer elaborating on the third sentence, perhaps "If you are somebody who wants to figure out how an organization such as the OCF gets and spends its money, what issues the OCF needs to deal with, and all the administrative work that needs to be done to keep the OCF running, please come to BoD meetings!"

I think it would be good to make a list of administrative things that the OCF needs help with to prospective staff, much like the list of technical issues below. I don't believe myself qualified enough to make such a list, but perhaps such a list might include the following.

Finances -
Revenue - include a link to the Fall 2018 buysheet and link some spreadsheet summarizing how our money is being used, and possible ways we may want to spend money on (desktop refresh, more servers, HPC, more stuff to lend)
Income - Briefly summarize (or copy from some previous slides) our funding sources, talk about referendum, etc. Talk about how the OCF needs people to come up with new ways to spend money, and new ways to get money.

Collaborations -
List past collaborations and invite reader to help develop our relationships with various other campus groups and tech companies (Facebook, Cloudless, plus whatever collaborations we've done with campus groups). Insert something like "We need people to look into other collaborations."

Lab maintenance -
We need people to help write and decide on a banning policy, CLA, and figure out how to provide more services that the majority can use other than just printing (perhaps lending stuff?). We need people to talk with <insert long list of roles that keep the OCF running>.

#### Staff Meeting

From 8 to 9 pm every Monday, all OCF staff (that includes you!) get together to
discuss technology, learn from each other, and work on OCF projects. These
meetings will be a mix of presentations, work time, and socializing. Our goal
is to build OCF staff into a strong community of capable individuals.

### Contribute

We know that finding ways to contribute to the OCF can feel daunting, but we
don’t want you to be discouraged. Don’t worry about finding the best way to
contribute, as long as you find something that you want to work on. Here are
some ways to get started:

#### Projects

We have a curated list of projects at [ocf.io/projects][projects]. Projects
labeled “newbie” are ones we think are best fit for new staffers, but we are
happy to provide context and mentorship on any project. In general, don’t worry
about fully understanding the project description. If you find a something that
interests you, respond to the thread or reach out to a veteran staffer to get
more information.

#### GitHub Issues

> All of our code is on GitHub (github.com/ocf). Sometimes you can find ways to

Reviewer: we should make github.com/ocf into a hyperlink

   * Install software packages on the desktops
     ([ocf_desktop/manifests/packages.pp][puppet-desktop-packages])
   * Configure Firefox to show the "are you sure you want to quit" dialog
     ([PR #373][puppet-373])
   * Allow operations staff to refund pages ([PR #368][puppet-368])
   * Monitor our mirrors to see if they’re out of date
     ([healthcheck script][mirror-healthcheck],
     [configuring server to call the healtcheck
     script][mirror-healthcheck-puppet],
     [Prometheus rules for alerting][prometheus-mirror])
   * Monitor printers to see if they’re out of paper, toner, or jammed
     ([printer.rules.yaml][prometheus-printer])
   * Send a desktop notification if a print jobs fails ([PR #321][puppet-321])
* [ocfweb][ocfweb] ([issues][ocfweb/issues]) - the source code for our website,
  www.ocf.berkeley.edu. Written in Python using the Django framework.
   * Hosts our documentation for users and staffers (including
     [this page][getinvolved-src]!) ([docs][docs-src])
   * JSON APIs for lab hours ([api/hours.py][api-hours])
   * [Account creation][account-register], [password resets][account-chpass],
     [others][account]

>    * Lab usage statistics ([stats/][ocfweb])

Reviewer: think you mean `[ocfweb-stats]`

>    * Form for requesting a virtual host ([lab_reservations/][lab_reservations])

Reviewer: wat

I think you forgot to update the text, also you probably want `[ocfweb-lab_reservations]`

   * Send automated emails ([misc/mail.py][ocflib-mail])
   * Printer querying and statistics ([printing/][ocflib-printing])
* [utils][utils] ([issues][utils/issues]) - similar to ocflib, this contains
  Python scripts that are run by humans.
   * Paper quota refund ([printing/paper][utils-paper])
   * Query/search OCF accounts ([acct/][utils-acct])
   * For users: create MySQL database and HTTP webdir
     ([makeservices/makemysql-real][utils-makemysql-real],
     [makeservices/makehttp][utils-makehttp])
   * Create new virtual machines ([staff/sys/makevm][utils-makevm])
   * Wake up desktops from sleep ([staff/lab/lab-wakeup][utils-lab-wakeup])
* [slackbridge][slackbridge] ([issues][slackbridge/issues]) - bridges our IRC
  network with our Slack organization, so staff members can exercise choice in
  what chat program they’d like to use. Written in Python.
* [ircbot][ircbot] ([issues][ircbot/issues]) - a fun chatbot, also used to
  approve new account requests. Written in Python.
   * See https://ircbot.ocf.berkeley.edu/ for a listing of IRC bot functionality

#### TODOs

If you’re feeling particularly adventurous, you can [search for the string

> “TODO” across our entire codebase][sourcegraph-todo].

Reviewer: WAIT NO WHERE DID THE "TODO: add another sentence here?" GO?

Author: rip wasn't sure if people would get the joke

#### Help requests

During the semester, our [[help@ocf.berkeley.edu mailing list|doc contact]]
gets around 14 emails per week from the UC Berkeley community. These emails get
responded to by volunteers like you. Most requests are about requesting virtual
hosting and helping users debug or fix their websites. If you’re interested in
seeing how this process works, ask a staffer to add you to the RT mailing list.
Since this is high-volume, we recommend filtering these to a different folder
so you don’t get notified all the time.

Joining this list doesn’t obligate you to respond to help requests. It’s
perfectly acceptable to only read tickets as a way to learn more about what we
do.

> If you’re on staff, you can view the ticket archive at rt.ocf.berkeley.edu.

Reviewer: rt.ocf.berkeley.edu should be a link

#### Your idea here

As a volunteer organization, the OCF’s direction is driven by the interests of
our members. If you have a new idea for the OCF, we want to help you build it.
You can use our chat channels or talk to a veteran staffer in person to get a
starting point and roadmap.

### Staff Hours

During the week, veteran staffers host “staff hours”
([ocf.io/staffhours][staffhours]), where we provide support to users of our
services. As a newbie, attending staff hours can serve multiple purposes:

   * Learn by shadowing veteran staffers as they help people out
   * Talk about your interests so we can help you find stuff to work on
   * Get one-on-one mentoring so you can learn about the vast complex world of
     OCF infrastructure
   * Get context and more information on a particular area you’re interested
     in contributing in

### Slack or IRC

OCF staff often communicate on Slack or IRC. For instructions on how to join

> Slack, see [ocf.io/slack][slack]. Some important channels:

Reviewer 3: perhaps note that these are bridged together, and maybe link IRC to thelounge?

Author: I'll just link to the docs

   * #rebuild - technical discussion
   * #administrivia - non-technical discussion
   * #ocf for off-topic, non-OCF related discussion

These channels are typically very active, and you shouldn’t feel the need to
read every line to stay updated. You also aren’t required to understand all the
technical jargon that’s used in #rebuild. Feel free to introduce yourself in
any of these channels if you’re new, and don’t be afraid to ask questions!

### Hang out in the lab

OCF staff members usually like to socialize in the lab. We’re a friendly bunch,

> so feel free to talk to us! You can also come into the lab after-hours-- just

Reviewer 3: emdash

say you’re interested in joining staff and we’ll let you in.

## FAQ

**I wasn’t able to make the first/second/nth meeting of the semester, can I
still join?**

> Yes! You can join at any time. If you’re worried about what you’ve missed, we

Reviewer 3: > worried about what you've missed
perhaps also mention they should feel free/comfortable to ask veteran staff to bring them up to speed

recommend reading archives from our mailing list ([ocf.io/announce][announce]).
These will usually contain links to presentations slides and information about
what was covered.

**What if I can’t physically be present at weekly meetings?**

You can still be part of OCF staff as the suggestions on this page are useful

> even if you can’t make meetings.

Reviewer: Feel like currently not quite enthusiastic enough. I'd reword to something like:
> You can still be a part of OCF staff! Even if you can't make meetings, the suggestions on this page are useful for getting started.

Reviewer 3: and perhaps mention that staying in touch via IRC or email is absolutely a way to stay 'in the community' even if they can't be physically present

﻿[[!meta title="Getting Involved"]]

## How can I get involved in the OCF?

Glad you asked! Students from all backgrounds are welcome to join staff. We
don’t interview or have a selective process for joining— you simply have to

> express interest and show up to our weekly meetings. If you want to learn more

Author: What do you think the expected time commitment is? I prefer to make it vague since we it varies a lot between staffers

> express interest and show up to our weekly meetings. If you want to learn more

Reviewer: >...you simply have to express interest and show up to our weekly meetings. **We don't have any time commitment expectations; some staff work many hours each week, and others only have time for occasional contributions.** If you want to learn more...

> express interest and show up to our weekly meetings. If you want to learn more

Author: sgtm, added a sentence that conveys this

be a mix of presentations, work time, and socializing. Our goal is to build OCF
staff into a strong community of capable individuals.

### Contribute

We know that finding ways to contribute to the OCF can feel daunting, but we
don’t want you to be discouraged. Don’t worry about finding the best way to
contribute, as long as you find something that you want to work on. Here are
some ways to get started:

#### Projects

We have a curated list of projects at [ocf.io/projects][projects]. Projects
labeled “newbie” are ones we think are best fit for new staffers, but we are
happy to provide context and mentorship on any project. In general, don’t worry
about fully understanding the project description. If you find a something that
interests you, respond to the thread or reach out to a veteran staffer to get
more information.

#### GitHub Issues

All of our code is on GitHub ([github.com/ocf][ocf-github]). Sometimes you can find ways to
contribute by reading open issues, but they can sometimes be hard to understand
since they’re short on context. Some issues are also marked as “good first
issue.” Here’s a list of our most active repos, with some examples of things we
can do with them:

> * [puppet][puppet] ([issues][puppet/issues]) - contains Puppet modules, which

Reviewer 4: Puppet is a tricky repo for devising examples because, in addition to actual configuration code, the Puppet repo also serves as a kind of software repository for various scripts which are only installed on a few hosts. If it were me, I would focus on the theme of configuration management, limiting examples to Puppet code and configuration files (as opposed to, for instance, trying to showcase the variety of files and their functions in this repository).

Some additional examples I think could be worth highlighting, that newcomers would be most likely to understand the significance of:
 - The Postfix file that configures our mail aliases
 - Our DHCP server config
 - Puppet code configuring Apache on our web server

Author: These examples look good, I'll add them. Though I still think the current examples are fine too, as long as there's a big variety I don't think people will get the wrong idea of what Puppet is about.

  controls the software and configuration on all of our servers.
   * Install software packages on the desktops
     ([ocf_desktop/manifests/packages.pp][puppet-desktop-packages])
   * Configure Firefox to show the "are you sure you want to quit" dialog
     ([PR #373][puppet-373])

>    * Allow operations staff to refund pages ([PR #368][puppet-368])

Reviewer 4: It's unlikely that newcomers to the OCF would understand why opstaff wouldn't otherwise be able to refund pages, or why this code belongs in the Puppet repo. I would find a different example.

   * Monitor our mirrors to see if they’re out of date
     ([healthcheck script][mirror-healthcheck],
     [configuring server to call the healtcheck
     script][mirror-healthcheck-puppet],
     [Prometheus rules for alerting][prometheus-mirror])
   * Monitor printers to see if they’re out of paper, toner, or jammed
     ([printer.rules.yaml][prometheus-printer])

>    * Send a desktop notification if a print jobs fails ([PR #321][puppet-321])

Reviewer 4: If I were a newcomer, reading through these three examples, I might think that Puppet's role is to monitor services or send notifications to desktops. These examples can be at the least reworded to emphasize what Puppet's role in enabling each of these is.

(For that matter, newcomers probably haven't heard of Prometheus fwiw)

Author: changed "Prometheus" to "Monitoring"

  Python scripts that are run by humans.
   * Paper quota refund ([printing/paper][utils-paper])
   * Query/search OCF accounts ([acct/][utils-acct])
   * For users: create MySQL database and HTTP webdir
     ([makeservices/makemysql-real][utils-makemysql-real],
     [makeservices/makehttp][utils-makehttp])
   * Create new virtual machines ([staff/sys/makevm][utils-makevm])
   * Wake up desktops from sleep ([staff/lab/lab-wakeup][utils-lab-wakeup])
* [slackbridge][slackbridge] ([issues][slackbridge/issues]) - bridges our IRC
  network with our Slack organization, so staff members can exercise choice in
  what chat program they’d like to use. Written in Python.
* [ircbot][ircbot] ([issues][ircbot/issues]) - a fun chatbot, also used to
  approve new account requests. Written in Python.
   * See https://ircbot.ocf.berkeley.edu/ for a listing of IRC bot functionality

#### TODOs

If you’re feeling particularly adventurous, you can [search for the string
“TODO” across our entire codebase][sourcegraph-todo].

#### Help requests

During the semester, our [[help@ocf.berkeley.edu mailing list|doc contact]]
gets around 14 emails per week from the UC Berkeley community. These emails get

> responded to by volunteers like you. Most requests are about requesting virtual

Reviewer 5: Sentence is awkwardly phrased. How about: "Volunteers like you respond to these emails"

hosting and helping users debug or fix their websites. If you’re interested in
seeing how this process works, ask a staffer to add you to the RT mailing list.
Since this is high-volume, we recommend filtering these to a different folder
so you don’t get notified all the time.

Joining this list doesn’t obligate you to respond to help requests. It’s
perfectly acceptable to only read tickets as a way to learn more about what we
do.

If you’re on staff, you can view the ticket archive at [rt.ocf.berkeley.edu][rt].

#### Your idea here

As a volunteer organization, the OCF’s direction is driven by the interests of
our members. If you have a new idea for the OCF, we want to help you build it.
You can use our chat channels or talk to a veteran staffer in person to get a
starting point and roadmap.

### Staff Hours

During the week, veteran staffers host “staff hours”
([ocf.io/staffhours][staffhours]), where we provide support to users of our

> services. As a newbie, attending staff hours can serve multiple purposes:

Reviewer 4: ```suggestion
services. Newbies are encouraged to attend staff hours too.
For a newbie, attending staff hours can serve multiple purposes:
```

﻿[[!meta title="Getting Involved"]]

## How can I get involved in the OCF?

Glad you asked! Students from all backgrounds are welcome to join staff. We

> don’t interview or have a selective process for joining—you simply have to

Reviewer 6: ```suggestion
don’t interview or have a selection process for joining—you simply have to
```

I think it is better to say we don't select rather than the process isn't selective (because that implies that there may be selection).

express interest and show up to our weekly meetings. We don't impose a specific
time commitment; some volunteers work many hours per week, others contribute
occasionally. If you want to learn more and be an active participant, there are
multiple ways to get started:

### Stay up-to-date

We use a mailing list ([ocf.io/announce][announce]) to announce weekly meetings
and recap past meetings. If you’re joining in the middle of the semester or

> missed a meeting, you can also read the mailing list archives to see what we

Reviewer 6: It'd be good to link to the archives here.

Author: ocf.io/announce is the archives

went over in the past. This is a low volume mailing list (2 emails per week)
and you can unsubscribe at any time.

[[Read more about other mailing lists available to join|doc
staff/mailing-lists]]

### In-person meetings

We hold two meetings every Monday:

> #### Board of Directors (BoD)

Reviewer 4: Put staff meeting before BoD, since it's more geared toward newcomers? (Or is this a deliberate move to encourage BoD participation?)

Author: I can see both sides, but I think it's best to put BoD first because it is held in time before staff meeting. It's less confusing this way.

Reviewer 4: On the other hand, though, in order of importance to newcomers, staff meeting would come before BoD...

don’t interview or have a selective process for joining—you simply have to
express interest and show up to our weekly meetings. If you want to learn more
and be an active participant, there are multiple ways to get started:

### Stay up-to-date

We use a mailing list ([ocf.io/announce][announce]) to announce weekly meetings
and recap past meetings. If you’re joining in the middle of the semester or
missed a meeting, you can also read the mailing list archives to see what we
went over in the past. This is a low volume mailing list (2 emails per week)
and you can unsubscribe at any time.

[[Read more about other mailing lists available to join|doc
staff/mailing-lists]]

### In-person meetings

We hold two meetings every Monday:

#### Board of Directors (BoD)

**Mondays 7-8pm**

BoD is where we discuss and vote on administrative decisions. If you are
someboday who wants to learn about how an organization such as the OCF gets and
spends its money, what issues the OCF needs to deal with, and all the
administrative work that needs to be done to keep the OCF running, please come
to BoD meetings! Meetings are always open to all. Some examples of what we

> discuss at BoD:

Reviewer: Most of these things happen outside of BoD, we just discuss progress at BoD (and sometimes vote on things). I kind of think we shouldn't group these examples under the "in-person meetings" category. However, I think this is definitely well-written and good for now. We can think about restructuring this part later.

Author: ahhh this is a draft, wasn't meant to go in this commit, see the gdocs list in #administrivia for comments on this

##### Finances

* Spending - we go through our "buy sheet" and vote on what to buy
* Income - we mainly get money from the ASUC and the Student Tech Fund. We
  discuss big-picture goals for receiving funding and where to direct our
  spending.

##### Collaborations

* Industry - we have mutually beneficial relationships with companies like Yelp,
  Facebook, Kloudless, and more.
* Campus groups - we want to improve in this area. Bring your ideas to BoD!

##### Lab Administration

> We draft policies and discuss new services to provide the UC Berkeley community.

Reviewer: imo this should be a bullet for consistency even though there's only one item
```suggestion
* We draft policies and discuss new services to provide the UC Berkeley community.
```
